### ISSUE TITLE
[a11y extension] documentation

### ISSUE THREAD

**Issue by pkra** [mathjax/MathJax-a11y#97]
Here's what I come up with.

* explain the two-in-one extensions or split them up
* document configuration options
* document menu options
* document behavior
  * global aria-label
  * walker 
  * collapse
  * interaction between the various parts, dependencies etc.

**Comment by dpvc** [comment]
I'm not sure what the "two-in-in" extension is.  Is it that the automatic collapsing is also included in this extension as well as the walking?

Would it be better to include only `Semantic-Complexity.js` (which I assume is needed by the walker) rather than `Semantic-Collapse.js`, and allow the page author to include `Semantic-Collapse.js` if desired?

**Comment by dpvc** [comment]
The `Semantic-Collapse` might also be part of the performance problem at start up.  As I recall, it goes through the various collapsings of the equation and records their widths, which means re-rendering the  equations several times.  This probably produces the "New Math" signal, and that may trigger the speech generation for each of those renderings?

**Comment by pkra** [comment]
> I'm not sure what the "two-in-in" extension is. Is it that the automatic collapsing is also included in this extension as well as the walking?

Yes, that's what I meant -- `Assistive-Explore.js` and `Semantic-Collapse.js` wrapped into one.

> Would it be better to include only Semantic-Complexity.js (which I assume is needed by the walker) rather than Semantic-Collapse.js, and allow the page author to include Semantic-Collapse.js if desired?

Assistive-Explore can now use Semantic-Collapse to provide a shorter summary on the top label (perhaps even as default, @zorkow?).

> This probably produces the "New Math" signal, and that may trigger the speech generation for each of those renderings?

@zorkow?

**Comment by dpvc** [comment]
> `Assistive-Explore `can now use `Semantic-Collapse` to provide a shorter summary on the top label 

I don't think it is `Semantic-Collapse` that it is using, but the results of `Semantic-Complexity`, which is why I suggested falling back to that.  @zorkow, are you just using the presence of the complexity mactions (which are created by Semantic-Complexity), or do you use the additional data about the widths of the various possible collapsings (added by Semantic-Collapse)?

**Comment by dpvc** [comment]
I'm wondering if the `Assistive-Explore` extension itself shouldn't load `Semantic-Complexity` if that is required by it (just like `Semantic-Complexity` forces `Semantic-MathML` and `Semantic-MathML` forces `sre_mathjax` to be loaded).  Then (given my comments in #96), the only thing left in `MathJax-accessibilty.js` would be the setting of the `[RespEq]` location (since `Semantic-MathML` sets `[SRE]` if it isn't already).

But in that case, I'd just recommend that page authors set `[RespEq]` themselves directly and load `Assistive-Explore` from there (and `Semantic-Collapse` if they want that as well).

The path to SRE in `Semantic-MathML` needs to be changed to correspond to the correct location (now that it will not be `processiveaccess`).

**Comment by zorkow** [comment]
Here are the answers for PR #92 
Summary is now produced by default as label, which decreases the lag on page load. I have not seen multiple New Math triggered, but I will check into that.

I was thinking about using complexities, but somehow they are always undefined in the DOM element, while set on MathML. So no, I don't. But I do need the mactions present in order to compute the summaries.

**Comment by zorkow** [comment]
Just for clarification:
mactions are set in Semantic-Complexity not Semantic-Collapse, right?
I need the former, not the latter.

**Comment by dpvc** [comment]
The mactions are set in Semantic-Complexity, not Semantic-Collapse, correct.

**Comment by dpvc** [comment]
The SRE code still seems to have [references to `progressiveaccess`](https://github.com/zorkow/speech-rule-engine/blob/master/src/common/system_external.js#L49-L50), unless that is changed in some other branch.

In any case, I'd like to see [line 39](https://github.com/zorkow/speech-rule-engine/blob/master/src/common/system_external.js#L39) check to see if the location is 'file:' and return false then, so that the explicit `https` version is used in line 50.

**Comment by dpvc** [comment]
Can the `sre.SystemExternal.url` use `Mathjax.Ajax.config.path.SRE` to find the URL for the JSON files?

**Comment by dpvc** [comment]
> I have not seen multiple New Math triggered, but I will check into that.

I also checked into that, and the re-rendering is being done at a lower level, so it doesn't trigger the signals.  That is good, so we don't have to worry about that.

**Comment by dpvc** [comment]
> I was thinking about using complexities, but somehow they are always undefined in the DOM element

I'm not sure they are properly marked to get output to the DOM elements.  And they are, I think they don't use the `data-semantic-` prefix, which they probably should.

**Comment by zorkow** [comment]
> Can the sre.SystemExternal.url use Mathjax.Ajax.config.path.SRE to find the URL for the JSON files?

Yes, I can overwrite that in the mathjax.js extension with the setup engine method. The feature vector for setting up the engine supports a json element.

**Comment by zorkow** [comment]
> I'm not sure they are properly marked to get output to the DOM elements. 
> And they are, I think they don't use the data-semantic- prefix, which they probably should.

Ah, all right. I thought attributes are generally pushed retained.

**Comment by pkra** [comment]
I've started to write https://github.com/mathjax/MathJax-RespEq/wiki/MathJax-Accessibility-Extensions-Documentation

**Comment by pkra** [comment]
I've completed a first pass for the documentation.

**Comment by dpvc** [comment]
Should we make edits there, or give you our comments for you to incorporate as you see fit?

One question:  is it really

    extensions: ["[Contrib]/Accessibility-Extension.js"]

or should it be

    extensions: ["[Contrib]/a11y/Accessibility-Extension.js"]

instead?  I have things set up in the extensions to use `[Contrib]/a11y/` because I thought that was the directory we agreed on last week.  Or did you really put them at the top level of the Contrib directory?

**Comment by pkra** [comment]
> Should we make edits there, or give you our comments for you to incorporate as you see fit?

Edits should be fine.

> did you really put them at the top level of the Contrib directory?

No, you are correct, it is / will be in `[Contrib]/a11y`.

**Comment by pkra** [comment]
@dpvc I've updated https://github.com/mathjax/MathJax-RespEq/wiki/MathJax-Accessibility-Extensions-Documentation.

**Comment by dpvc** [comment]
@pkra, I made an editing pass.  When I look at it in print, I like `auto-collapse.js` better than `auto collapse.js` after all.  Can we change to that before I make the edits to the files themselves?

**Comment by dpvc** [comment]
@pkra, you mention in the document that the explorer disables `AssisitveMML`, but I don't think that is true currently.  Do I need to add that?

**Comment by pkra** [comment]
> Do I need to add that?

Yes, please. Otherwise, mathml- capable AT will read both the label and the hidden MathML.

**Comment by pkra** [comment]
Since we can't ping the issue from the wiki: latest diff https://github.com/mathjax/MathJax-RespEq/wiki/MathJax-Accessibility-Extensions-Documentation/_compare/2dec65468cfaf498635109ddd6413d0a28cdac32...2c4b08057d4190a92c6262c9137fb1c1ca3ecb57

**Comment by pkra** [comment]
A three more fixes https://github.com/mathjax/MathJax-RespEq/wiki/MathJax-Accessibility-Extensions-Documentation/_compare/2c4b08057d4190a92c6262c9137fb1c1ca3ecb57...12dacd731fd8e21ddc87ae0ab301adb566d563fd

**Comment by pkra** [comment]
Closing this for now.
